### ISSUE TITLE
Invalid syntax with walrus and yield not detected

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#5709]
**Describe the bug**
Using an assignment expression inside of a yield doesn't work at runtime but pyright is fine with it

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
for news_item in msg.news:
    yield news := UserNews(self._state, news_item, achievements.get(news_item.gameid, {}))
```

<img width="805" alt="image" src="https://github.com/microsoft/pyright/assets/50501825/fe67c3da-18fb-4bf4-b3df-4431ffe20898">
Running the file gives the error (you need brackets)

```py
    yield news := UserNews(self._state, news_item, achievements.get(news_item.gameid, {}))
               ^^
SyntaxError: invalid syntax
```

**VS Code extension or command-line**
pylance v2023.8.20

**Comment by erictraut** [comment]
Here's a minimal self-contained repro:
```python
def func():
    yield news := 1
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
